### PR TITLE
Updated/modernized tests. Made HamiltonianProblem() consistent

### PR DESCRIPTION
**Pull request by ChrisDeGrendele**

### CODE REVIEW

File: src/hamiltonian.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@ end
 
 struct PhysicsTag end
 
-function HamiltonianProblem{T}(H,p0,q0,tspan,p=nothing;kwargs...) where T
+function HamiltonianProblem{T}(H,q0,p0,tspan,p=nothing;kwargs...) where T
```
Comment (SebastianM-C) [inline src/hamiltonian.jl]: I think the `p,q` order wast intentionally set like this in order to be consistent with how the rest of the problems are defined. See https://sciml.ai/2018/01/24/Parameters.html for more details.

Comment (ChrisDeGrendele) [reply src/hamiltonian.jl]: Why is line 3 the opposite then?

Comment (SebastianM-C) [reply src/hamiltonian.jl]: I think it was missed during the update. You can see the rest of the changes in [this](https://github.com/SciML/DiffEqPhysics.jl/commit/65c93463a818e3538851770ea69cf1849b0f458f) commit by @ChrisRackauckas.

**Comment by ChrisDeGrendele** [comment]
How is that ^ ? I didn't update the other tests because they all went to NBodySimulation.jl. I believe @ChrisRackauckas said to just remove them? Is that all the NBodyStuff?

**Comment by SebastianM-C** [comment]
[The test fail](https://travis-ci.org/github/SciML/DiffEqPhysics.jl/jobs/673082120#L453) because the SafeTestsets package is not in the Project.toml
To solve this, you need to add the SafeTestsets package as a test dependency of DiffEqPhysics.

One way of doing this would be to activate the environment of the DiffEqPhysics package (`] activate DiffEqPhysics`) and
```
] add SafeTestsets
```
Then, inspect the Project.toml file of DiffEqPhysics and move SafeTestsets to `[extras]` and add it to `[targets]`.

**Comment by ChrisDeGrendele** [comment]
Any other comments?  @SebastianM-C @ChrisRackauckas (Just a bump, thank you both for the help :) )

**Pull request merged**